x-pack/plugins/security_solution/public/common/hooks/eql/use_eql_preview.ts

```diff
@@ -3,10 +3,145 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { noop } from 'lodash/fp';
+import { Subscription } from 'rxjs';
 
-import { useAsync, withOptionalSignal } from '../../../shared_imports';
-import { getEqlPreview } from './api';
+import * as i18n from '../translations';
+import { useKibana } from '../../../common/lib/kibana';
+import {
+  AbortError,
+  isCompleteResponse,
+  isErrorResponse,
+} from '../../../../../../../src/plugins/data/common';
+import {
+  EqlSearchStrategyRequest,
+  EqlSearchStrategyResponse,
+} from '../../../../../data_enhanced/common';
+import { getEqlAggsData, getSequenceAggs } from './helpers';
+import { EqlPreviewResponse, EqlPreviewRequest, Source } from './types';
+import { hasEqlSequenceQuery } from '../../../../common/detection_engine/utils';
+import { EqlSearchResponse } from '../../../../common/detection_engine/types';
+import { parseScheduleDates } from '../../../../common/detection_engine/parse_schedule_dates';
+import { inputsModel } from '../../../common/store';
+import { EQL_SEARCH_STRATEGY } from '../../../../../data_enhanced/public';
 
-const getEqlPreviewWithOptionalSignal = withOptionalSignal(getEqlPreview);
+export const useEqlPreview = (): [
+  boolean,
+  (arg: EqlPreviewRequest) => void,
+  EqlPreviewResponse
+] => {
+  const { data, notifications } = useKibana().services;
+  const refetch = useRef<inputsModel.Refetch>(noop);
+  const abortCtrl = useRef(new AbortController());
+  const [loading, setLoading] = useState(false);
+  const didCancel = useRef(false);
 
-export const useEqlPreview = () => useAsync(getEqlPreviewWithOptionalSignal);
+  const [response, setResponse] = useState<EqlPreviewResponse>({
+    data: [],
+    inspect: {
+      dsl: [],
+      response: [],
+    },
+    refetch: refetch.current,
+    totalCount: 0,
+  });
+
+  const searchEql = useCallback(
```

Comment:
I'm not sure if it's the implementation here or something else, but I see a behavioral difference between the EQL preview loading and the other query previews loading. With EQL it does not populate the histogram with the loader: 

![Oct-08-2020 16-58-13](https://user-images.githubusercontent.com/657252/95518225-42738000-0988-11eb-89ac-b6a117caecbd.gif)

Reply:
Yea, I wanted to try to keep the PRs smallish - so I focused on fixing the bugs/tests with the api here and have a sequel "preview_buggies_2" that addresses these issues (https://github.com/elastic/kibana/pull/80110). I promise it's a really good sequel 😄 

It currently includes the changes from this PR, but if you check out the last 2 commits you can see the new changes.